PowerSync/PowerSync.Common/Client/PowerSyncDatabase.cs

```diff
@@ -231,21 +232,34 @@ private async Task LoadVersion()
         }
     }
 
-    private record LastSyncedResult(string? synced_at);
+    private record LastSyncedResult(int? priority, string? last_synced_at);
 
     protected async Task UpdateHasSynced()
     {
-        var result = await Database.Get<LastSyncedResult>("SELECT powersync_last_synced_at() as synced_at");
+        var results = await Database.GetAll<LastSyncedResult>(
+            "SELECT priority, last_synced_at FROM ps_sync_state ORDER BY priority DESC"
+        );
 
-        var hasSynced = result.synced_at != null;
-        DateTime? syncedAt = result.synced_at != null ? DateTime.Parse(result.synced_at + "Z") : null;
+        DateTime? lastCompleteSync = null;
 
+        foreach (var result in results)
```

Comment:
The logic here looks a little strange without processing the results for all the sync priorities. Specifically: querying the results for all the priorities then looping through the list of results to find the `FULL_SYNC_PRIORITY` row. At face value that seems a bit strange (we could just be filtering the query result).

I'd suggest adding a comment that this is incomplete and will be refined when reporting individual sync priority statuses in the SyncStatus.